build broken on some config because of "field ‘skc_linux_alloc’ has incomplete type"

### System information
Type | Version/Name
 --- | ---
Distribution Name	| L4T (Linux 4 Tegra, based on Ubuntu)
Distribution Version	| 32.4.3
Linux Kernel	| 4.9.140-tegra
Architecture	| aarch64
ZFS Version	| c15d36c674bcfb10975fc835978d4a49d159bf0b
SPL Version	| N/A

### Describe the problem you're observing

This is on a Jetson TX1; build fails with:
```
In file included from /scratch/dolbeau/zfs-upstream/module/spl/../os/linux/spl/spl-kmem-cache.c:26:0:
/scratch/dolbeau/zfs-upstream/include/os/linux/spl/sys/kmem_cache.h:203:24: error: field ‘skc_linux_alloc’ has incomplete type
  struct percpu_counter skc_linux_alloc;   /* Linux-backed Obj alloc  */
```

Other files also fail to compile, apparently if they have  `#include <sys/kmem_cache.h>`but not `#include <linux/percpu_compat.h>` before it.

### Describe how to reproduce the problem

Build latest GIT on the same configuration...

### Known fix

Moving the include earlier in the sequence in spl-kmem-cache.c & adding to other solves the problem (see attached patch)

[patch.txt](https://github.com/openzfs/zfs/files/4919537/patch.txt)

--- a/module/os/linux/spl/spl-kmem-cache.c
+++ b/module/os/linux/spl/spl-kmem-cache.c
@@ -22,6 +22,7 @@
  *  with the SPL.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <linux/percpu_compat.h>
 #include <sys/kmem.h>
 #include <sys/kmem_cache.h>
 #include <sys/shrinker.h>
@@ -31,7 +32,6 @@
 #include <sys/wait.h>
 #include <linux/slab.h>
 #include <linux/swap.h>
-#include <linux/percpu_compat.h>
 #include <linux/prefetch.h>
 
 /*

--- a/module/os/linux/spl/spl-vmem.c
+++ b/module/os/linux/spl/spl-vmem.c
@@ -22,6 +22,7 @@
  *  with the SPL.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <linux/percpu_compat.h>
 #include <sys/debug.h>
 #include <sys/vmem.h>
 #include <sys/kmem_cache.h>

--- a/module/os/linux/spl/spl-zlib.c
+++ b/module/os/linux/spl/spl-zlib.c
@@ -54,6 +54,7 @@
  */
 
 
+#include <linux/percpu_compat.h>
 #include <sys/kmem.h>
 #include <sys/kmem_cache.h>
 #include <sys/zmod.h>

